tests/lib/rules/jsx-child-element-spacing.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const rule = require('../../../lib/rules/jsx-child-element-spacing');
+const RuleTester = require('eslint').RuleTester;
+const parserOptions = {
+  sourceType: 'module',
+  ecmaFeatures: {
+    jsx: true
+  }
+};
+
+const ERROR_MESSAGE = [{message: 'Ambiguous spacing between child elements.'}];
+
+const ruleTester = new RuleTester({parserOptions});
+ruleTester.run('jsx-child-element-spacing', rule, {
+  valid: [{
+    code: `
+      <App>
+        foo
+        bar
+      </App>
+    `
+  }, {
+    code: `
+      <App>
+        <a>foo</a>
+        <a>bar</a>
+      </App>
+    `
+  }, {
+    code: `
+      <App>
+        foo<a>bar</a>baz
+      </App>
+    `
+  }, {
+    code: `
+      <App>
+        foo
+        {' '}
+        <a>bar</a>
+        {' '}
+        baz
+      </App>
+    `
+  }, {
+    code: `
+      <App>
+        foo{/*
+        */}<a>bar</a>{/*
+        */}baz
+      </App>
+    `
+  }],
+
+  invalid: [{
+    code: `
+      <App>
+        foo
+        <a>bar</a>
```

Comment:
what would the output be if you wanted this autofixed?

Reply:
I think as you pointed out in the original issue, the intent is ambiguous so autofixing may not be feasible.

If we were to autofix, replacing with the comment form would maintain the semantic without too much disruption (though would perhaps be less aesthetically pleasing)

I would advocate for no automatic action since, in my experience, the developer most often intended to add a space, though I could be convinced otherwise